pkg/network/network.go

```diff
@@ -136,6 +134,72 @@ func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, opensh
 	return objs, nil
 }
 
+// RenderObjsToRemove creates list of components to be removed
+func RenderObjsToRemove(prev, conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
+	log.Print("starting rendering objects to delete phase")
+	objsToRemove := []*unstructured.Unstructured{}
+
+	if prev == nil {
+		return objsToRemove, nil
+	}
+
+	if conf.Multus == nil {
+		o, err := renderMultus(prev, manifestDir, openshiftNetworkConfig, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.LinuxBridge == nil {
+		o, err := renderLinuxBridge(prev, manifestDir, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.KubeMacPool == nil {
+		o, err := renderKubeMacPool(prev, manifestDir)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.NMState == nil {
+		o, err := renderNMState(prev, manifestDir, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	if conf.Ovs == nil {
+		o, err := renderOvs(prev, manifestDir, clusterInfo)
+		if err != nil {
+			return nil, err
+		}
+		objsToRemove = append(objsToRemove, o...)
+	}
+
+	// Remove operator namespace occurences, since its not allowed to remove it, because cnao pod is there.
+	// TODO cleanup OPERAND_NAMESPACE once there are no components using it.
+	operatorNamespace := os.Getenv("OPERATOR_NAMESPACE")
+	index := 0
+	for _, obj := range objsToRemove {
+		name := obj.GetName()
+		if name != operatorNamespace {
```

Comment:
You cannot just check for the Name, that could remove any object with the same name. Please add a check for the object type. If it is Namespace with the operand namespace name, don't remove it.